health-sdk/sdk/src/main/java/net/gini/android/health/sdk/review/ReviewViewModel.kt

```diff
@@ -81,9 +81,10 @@ internal class ReviewViewModel(val giniHealth: GiniHealth, val paymentProviderAp
                     val newEmptyValidationMessages =
                         paymentDetails.validate().filterIsInstance<ValidationMessage.Empty>()
 
-                    // Clear IBAN error, if IBAN changed
+                    // Clear IBAN error, if IBAN changed, and check validation for new iban
                     if (prevPaymentDetails != null && prevPaymentDetails.iban != paymentDetails.iban) {
                         nonEmptyValidationMessages.remove(ValidationMessage.InvalidIban)
+                        nonEmptyValidationMessages.addAll(validateIban(paymentDetails.iban).filterIsInstance<ValidationMessage.InvalidIban>())
```

Comment:
[Clarification] I am interested in why you added the IBAN revalidation here. I saw it caused two tests to fail so we intentionally did not revalidate the IBAN here:
![Screenshot 2024-03-05 at 18 32 54](https://github.com/gini/gini-mobile-android/assets/13149498/2c3afaa3-fed9-4e86-af16-6f17a99698ea)

I checked iOS and there validation is also repeated only on pay button press. So we need to have a good reason to change this behaviour on Android.

Reply:
Hi, I saw in this comment https://ginis.atlassian.net/browse/IPC-160?focusedCommentId=23915 that the IBAN validation error appears when the field is no longer in focus - this was not happening when I tested, this being the reason I added the revalidation. 

I can take it out and leave it as is so we match iOS implementation and keep the tests from failing.